x/evm/keeper/fees.go

```diff
@@ -55,8 +58,23 @@ func (k *Keeper) DeductTxCostsFromUserBalance(
 		return errorsmod.Wrapf(err, "account not found for sender %s", from)
 	}
 
-	// deduct the full gas cost from the user balance
-	if err := authante.DeductFees(k.bankKeeper, ctx, signerAcc, fees); err != nil {
+	// Convert the balance to 18 decimals
+	// TODO: Replace base Denom with ATOM
+	signerBalance := k.bankKeeper.GetBalance(ctx, signerAcc.GetAddress(), utils.BaseDenom)
+	balanceIn18Decimals := utils.ConvertTo18Decimals(*signerBalance.Amount.BigInt())
```

Comment:
same here, how do you know if the denom has already 18 decimals or not

Reply:
I believe here we are assuming gas cost will be paid in ATOM which is always 6 decimals